server/api/image/image.controller.js

```diff
@@ -1,39 +1,164 @@
 const fs = require('fs-extra')
+const path = require('path')
 const cuid = require('cuid')
 const slug = require('slug')
 const { config } = require('../../../config/config')
 const { cloudUploadService } = require('./cloudImageUpload')
-// any depended upon api services
+const { resizeImage } = require('../../services/image/imageResize')
+
+/**
+ * A list of predefined usages of images and their automatically generated sizes.
+ *
+ * @type {Map<String, {
+    name: string,
+    width: Number,
+    height: Number
+  }[]>}
+ */
+const _usages = new Map()
+
 const uploadImage = async (req, res) => {
   try {
-    const ImageBin = req.body.image
-    const ImageBuffer = Buffer.from(ImageBin, 'binary')
-    const uniqueID = cuid()
-    slug.defaults.mode = 'rfc3986'
-    const uploadUrl = `/static/upload${(process.env.NODE_ENV === 'test') ? '-test' : ''}`
-    const uploadPath = `./public${uploadUrl}`
-    !fs.existsSync(uploadPath) && fs.mkdirSync(uploadPath)
-    const filename = `${uploadUrl}/${uniqueID}-${slug(req.body.file)}`
-    const fqp = `./public${filename}`
+    const { image: imageBin, file: originalFilename } = req.body
+    const imageBuffer = Buffer.from(imageBin, 'binary')
+
+    // 1. Store the original file - do this so we can always go back and re-process the raw file.
```

Comment:
True, it won't have any direct use on the website after it's uploaded, but just in case we want to retrieve the original file and do something with it. Debatable whether it's worth keeping the file or not

Reply:
keep for now - revisit later